frontend/src/components/viewers/Tensorboard.tsx

```diff
@@ -235,6 +253,25 @@ class TensorboardViewer extends Viewer<TensorboardViewerProps, TensorboardViewer
     return urls.length === 1 ? urls[0] : urls.map((c, i) => `Series${i + 1}:` + c).join(',');
   }
 
+  private async _checkTensorboardPodStatus(): Promise<void> {
+    // If pod address is not null and tensorboard pod doesn't seem to be read, pull status again
+    if (this.state.podAddress && !this.state.tensorboardReady) {
+      fetch('apis/v1beta1/_proxy/' + this.state.podAddress.replace(/(^\w+:|^)\/\//, ''), {
+        method: 'HEAD',
+      })
+        .then(res => {
+          if (res.status === 200) {
+            this.setState({ tensorboardReady: true });
+          } else {
+            this.setState({ tensorboardReady: false });
```

Comment:
Recommend changing it to
```suggestion
            this.setState(({ tensorboardReady }) => ({ tensorboardReady: tensorboardReady || false }));
```
because network requests may come back in different order than they are sent, maybe we have already detected tensorboard is ready.

Reply:
Done